pygsuite/drive/file.py

```diff
@@ -0,0 +1,231 @@
+        client = client or Clients.drive_client
+
+        # get upload file size
+        filesize = os.path.getsize(filepath)
+
+        # establish if the upload should be resumable
+        # TODO: expand upon this and determine how this should work for users
+        if filesize > DRIVE_FILE_MAX_SINGLE_UPLOAD_SIZE:
+            resumable = True
+        else:
+            resumable = False
+
+        # get filename and extension
+        _, extension = os.path.splitext(filepath)
+
+        # name of the file in Drive
+        name = name if name else os.path.basename(filepath)
+
+        if convert_to:
+            # try to coerce str into a GoogleDocFormat
+            if isinstance(convert_to, str):
+                try:
+                    convert_to = GoogleDocFormat[convert_to.upper()]
+                except Exception as e:
+                    raise ValueError(
+                        f"For converting to a Google Document, please use one of the following inputs:\n{[item.name for item in GoogleDocFormat]}"
+                    ) from e
+
+            # find the corresponding mime type of the extension of the upload file
+            try:
+                mimetype = FILE_MIME_TYPE_MAP[convert_to][extension.lower()]
+            except Exception as e:
+                raise ValueError(
+                    f"File extension {extension.lower()} is not supported with the Google Document type {convert_to.value}"
+                ) from e
+
+        media_body = MediaFileUpload(
+            filename=filepath, mimetype=mimetype, chunksize=-1, resumable=resumable
+        )
+
+        file = cls.create(
+            name=name,
+            mimetype=mimetype,
+            media_body=media_body,
+            client=client,
+        )
+
+        return file
+
+    def fetch_metadata(
+        self,
+        fields: List[str] = ["id", "kind", "name", "mimeType"],
+        cache: bool = True,
+    ) -> dict:
+        """Metadata for the file, based on the files.get method.
+        Default fields include kind, name, and mimetype.
+        Additional fields available are found here:
+        https://googleapis.github.io/google-api-python-client/docs/dyn/drive_v3.files.html#get
+
+        Args:
+            cache (bool): whether to first look at the cached metadata.
+        """
+
+        # see if we have cached the file metadata already
+        if (self._metadata is None or cache is False):
+
+            self._metadata = (
+                self.client.files()
+                .get(
+                    fileId=self.id,
+                    fields=f"{', '.join(fields)}",
+                )
+            ).execute()
+
+        return self._metadata
+
+    @property
+    def kind(self):
+
+        return self.fetch_metadata().get("kind")
+
+    @property
+    def name(self):
+
+        return self.fetch_metadata().get("name")
+
+    @property
+    def mimetype(self):
+
+        return self.fetch_metadata().get("mimeType")
+
+    def share(self):
```

Comment:
should be able to copy this from the DriveObject

Reply:
Yeah, when I have my next chance to spend some good time on this, my plan was to merge other methods / ideas from the `DriveObject` that aren't yet copied here, such as `share`, `comments`, etc.